server/data/exampleApiClient.ts

```diff
@@ -0,0 +1,41 @@
+import {
+  RestClient,
+  TokenStore,
+  getSystemClientTokenFromHmppsAuth,
+  asSystem,
+} from '@ministryofjustice/hmpps-rest-client'
+import config from '../config'
+import logger from '../../logger'
+
+export default class ExampleApiClient extends RestClient {
+  constructor(tokenStore: TokenStore) {
+    super('Example API', config.apis.exampleApi, logger, (username: string) =>
+      getSystemClientTokenFromHmppsAuth(config.apis.hmppsAuth, tokenStore, logger, username),
```

Comment:
With hindsight, it might be better to wrap this in a class and explicitly encourage it to be reused. 

```typescript
const hmppsAuthTokenClient = new HmppsAuthTokenClient(config.apis.hmppsAuth, tokenStore, logger)
const exampleApiClient = new ExampleApiClient(hmppsAuthTokenClient)
```
which would then change this to: 
```typescript
constructor(hmppsAuthTokenClient: HmppsAuthTokenClient) {
    super('Example API', config.apis.exampleApi, logger, hmppsAuthTokenClient)
}
```
...
```typescript
 const token = hmppsAuthTokenClient.getToken(username)
```

Reply:
Cool with me - I think I'll make a `AuthenticationClient` interface or something that needs to implement a `getToken` endpoint, then I'll build a HMPPS Auth one and include that in the template

Reply:
Actioned in https://github.com/ministryofjustice/hmpps-template-typescript/pull/522/commits/3079913518dd718504ff4c3318bc587da8b32748

Reply:
On another note, breaking this into the `HmppsAuthTokenClient` has given a quite clear option for breaking this out into it's own `hmpps-typescript-lib` module